Profile post

andy-abreu opened the pull request:
Finished the post mechanism for the profile page. Also integrated prickle and ajax.

andy-abreu committed:
Added a simple post mechanism for the profile page

andy-abreu committed:
Added json to the post mechanism of the profile page

andy-abreu committed:
Fixed the 'get' potential issue on the Unpickle call in ProfilePage. Also fixed a minor bug with the alert banners in the profile page

rjkernick commented on client/src/main/scala/com/tripPlanner/webapp/pages/ProfileJsImpl.scala:
```diff
@@ -1,22 +1,27 @@
   * Created by rjkj on 12/5/15.
   */
-object ProfileJsImpl extends ProfileJs {
+class ProfileJsImpl extends ProfileJs {
```
I would make this a trait

rjkernick commented on client/src/main/scala/com/tripPlanner/webapp/pages/ProfileJsImpl.scala:
```diff
@@ -30,43 +35,52 @@ object ProfileJsImpl extends ProfileJs {
+                  option(value := "Mountain Time (US & Canada)")("(GMT-07:00) Mountain Time (US & Canada)"),
+                  option(value := "Central Time (US & Canada)")("(GMT-06:00) Central Time (US & Canada)"),
+                  option(value := "Eastern Time (US & Canada)", selected := "selected")("(GMT-05:00) Eastern Time (US & Canada)"),
                   option(value := "Indiana (East)")("(GMT-05:00) Indiana (East)")
```
Should add a technical task to populate this list from the db or at least the server side in the future

Commit:
Added route for bootstrap fonts

rjkernick committed:
:Merge branch 'profile-post' of https://github.com/rjkernick/TripPlanner into profile-post

rjkernick committed:
Fixed Url

rjkernick committed:
Added glyph

andy-abreu committed:
Switched the ProfileJsImpl from class to object and also added some glyph-icons
also removed the post result data display from the page.

andy-abreu committed:
Merge branch 'logging' of https://github.com/rjkernick/TripPlanner into profile-post

Conflicts:
	client/src/main/scala/com/tripPlanner/webapp/pages/ProfileJsImpl.scala
	server/src/main/scala/com/tripPlanner/webapp/Routes.scala
	server/src/main/scala/com/tripPlanner/webapp/pages/ProfilePage.scala

andy-abreu committed:
Played with flyway and had to do a few changes under the db.migration folder.
It seemed not to have liked having separate files so I put the creates under one file for now.

andy-abreu committed:
Added a separate file to create the tables used by the profile mechanism

andy-abreu committed:
Initial commit... working with prickle and nested classes

rjkernick committed:
Added domain dependency to server
Changed Tables.scala and updated UserDao

andy-abreu committed:
Had to modified the package name under shared in order to prevent classpath confusion in between packages

andy-abreu committed:
Merge branch 'profile-post' of https://github.com/rjkernick/TripPlanner into profile-post

Conflicts:
	domain/src/main/scala/com/tripPlanner/domain/UserDao.scala

rjkernick committed:
Added DomainSupport

rjkernick committed:
Merge remote-tracking branch 'origin/dev' into profile-post

rjkernick committed:
Fix imports

andy-abreu committed:
The profile page now saves to the DB
State dropdown populates from the StateDao (mocked for now since I couldn't
figure out how to do the select)
flyway populates state table static data

rjkernick committed:
Added query for states

rjkernick commented on domain/src/main/resources/db/migration/V2__create_profile_tables.sql:
```diff
@@ -0,0 +1,27 @@
+create table ADDRESS (
+    USER_ID varchar(50) not null,
```
Should make User_id a foreign key and add ID field

rjkernick commented on domain/src/main/resources/db/migration/V2__create_profile_tables.sql:
```diff
@@ -0,0 +1,27 @@
+);
+
+create table VEHICLE (
+    USER_ID varchar(50) not null,
```
Same here

andy-abreu committed:
Made a few fixes based on suggestions on pull request #5

andy-abreu committed:
Merge branch 'dev' into profile-post

Conflicts:
	client/src/main/scala/com/tripPlanner/webapp/App.scala
	domain/src/main/scala/com/tripPlanner/domain/UserDao.scala
	server/src/main/scala/com/tripPlanner/webapp/pages/ProfilePage.scala

andy-abreu committed:
Modified the DAOs to make a differentiation between create and save (update)

rjkernick committed:
Dao fixed

andy-abreu committed:
Added the static inserts required to populate the state table in the DB

andy-abreu committed:
The profile page is now split into three different sections, personal info, vehicle info and Change password sections

rjkernick committed:
Updates to dao

rjkernick committed:
Finished updates to dao, added some test cases

andy-abreu committed:
Removed all the ajax stuff from the multi-form profile page.
Also added all the ground work needed to add multiple addresses and
multiple vehicles on the profile

andy-abreu committed:
Modified the addressDao to get the address list by userId

andy-abreu committed:
Added userDao.getById

andy-abreu committed:
Added the dynamic text fields for handling multiple addresses and multiple vehicles on the profile forms

rjkernick committed:
updates to js pages

rjkernick committed:
Added NavPill trait to centralize navigation
